src/signature.js

```diff
@@ -71,36 +70,53 @@ angular.module('signature').directive('signaturePad', ['$window', '$timeout',
         }
       ],
       link: function (scope, element, attrs) {
-        canvas = element.find('canvas')[0];
-        scope.signaturePad = new SignaturePad(canvas);
+        var canvas = element.find('canvas')[0];
+        var parent = canvas.parentElement;
+        var ctx = canvas.getContext('2d');
+
+        var width = parseInt(scope.width, 10);
+        var height = parseInt(scope.height, 10);
+        var aspectRatio = height / width;
+
+        canvas.width = width;
+        canvas.height = height;
         
         //Resize to fix UI Bootstrap Modal Show problem
         $timeout(function(){
```

Comment:
The timeout is not sufficient. Please see what happens when size of the `div.signature` is changed for any other reason than resizing the window.

We should continuously check if the size of the div has changed. For instance using `$interval` rather than `$timeout`.